golem-worker-executor-base/src/durable_host/filesystem/types.rs

```diff
@@ -75,16 +75,28 @@ impl<Ctx: WorkerCtx> HostDescriptor for DurableWorkerCtx<Ctx> {
         length: Filesize,
         advice: Advice,
     ) -> Result<(), FsError> {
+        let _permit = self
```

Comment:
Ditto for this comment. All these async permits will add overhead and boilerplate and are difficult to enforce.